src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -1277,11 +1277,18 @@ static String createMergeQuery(TableId targetTableId, List<String> primaryKeys,
   }
 
   private void runWithRetries(ContextualRunnable runnable, long retryDelay, String dataset, String schema, String table,
-    String onFailedAttemptMessage, String retriesExhaustedMessage) throws DeltaFailureException, InterruptedException {
+                              String onFailedAttemptMessage, String retriesExhaustedMessage) throws DeltaFailureException, InterruptedException {
     RetryPolicy<Object> retryPolicy = createBaseRetryPolicy(retryDelay)
       .onFailedAttempt(failureContext -> {
         Throwable t = failureContext.getLastFailure();
         LOG.error(onFailedAttemptMessage, t);
+        //Logging error list
+        if (t instanceof BigQueryException) {
+          List<BigQueryError> errors = ((BigQueryException) t).getErrors();
+          if (errors != null) {
+            errors.forEach(err -> LOG.error(err.getMessage()));
```

Comment:
BigQueryError has few other properties than message - reason, location
Would it make sense to log those too?

Reply:
It might not always be useful and relevant.This info added in such cases creates confusion.Error message is pretty much clear to take action.